library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -733,6 +814,9 @@ public void onLoadCompleted(Chunk loadable, long elapsedRealtimeMs, long loadDur
   @Override
   public void onLoadCanceled(
       Chunk loadable, long elapsedRealtimeMs, long loadDurationMs, boolean released) {
+    if (interrupted) {
+      discardAfterInterrupt();
```

Comment:
Can you just move this into the "if (interrupted)" condition below? Seems more logical to have it all in one place.